src/libcrun/handlers/wasmedge.c

```diff
@@ -118,6 +120,32 @@ libwasmedge_exec (void *cookie, __attribute__ ((unused)) libcrun_container_t *co
   WasmEdge_ConfigureAddProposal (configure, WasmEdge_Proposal_ReferenceTypes);
   WasmEdge_ConfigureAddProposal (configure, WasmEdge_Proposal_SIMD);
   WasmEdge_ConfigureAddHostRegistration (configure, WasmEdge_HostRegistration_Wasi);
+  // Check if the environment variable is set
+  const char *nnPreloadEnv = getenv ("WASMEDGE_WASINN_PRELOAD");
+  if (nnPreloadEnv != NULL)
+    {
+      char *nnPreloadEnvCopy = xstrdup (nnPreloadEnv);
```

Comment:
you could define it as `cleanup_free char *nn_preload_copy = xstrdup (nn_preload_env);`

and do not need to worry about freeing it.